src/apiserver/api_server_impl.h

```diff
@@ -92,6 +94,9 @@ Archiver& operator&(Archiver& ar, QueryReq& s) {  // NOLINT
     // mode is not optional
     ar.Member("mode") & s.mode;
     ar.Member("sql") & s.sql;
+    if (ar.HasMember("parameters")) {
```

Comment:
JsonWriter HasMember is always false. If you don't want to support write `QueryReq ` to json, delete the template, just impl JsonReader reads QueryReq